docs/source/common_issues.rst

```diff
@@ -821,3 +821,55 @@ This is best understood via an example:
 To get this code to type check, you could assign ``y = x`` after ``x`` has been
 narrowed, and use ``y`` in the inner function, or add an assert in the inner
 function.
+
+
+Upgrading to :pep:`585` syntax
+------------------------------
+
+In pre-:pep:`585` syntax type annotations for builtin types such as ``dict``,
+``list``, ``tuple``, ``type`` were taken from the :py:mod:`typing` module as
+``typing.Dict``, ``typing.List``, ``typing.Tuple``, ``typing.Type`` etc.
+
+When updating a codebase to use modern syntax, you may hit a case such as the following:
+
+.. code-block:: python
+
+    from typing import TypeVar
+
+    T = TypeVar("T", bound="Foo")
+
+    class Foo:
+	type: str
+
+	@classmethod
+	def bar(cls: type[T], value: str) -> None:
+	    pass    # Mypy complains that "Foo.type" is "not valid as a type"
```

Comment:
I'd put the comment above the method signature rather than below it.

```suggestion
        # Mypy complains that "Foo.type" is "not valid as a type"
	@classmethod
	def bar(cls: type[T], value: str) -> None:
	    pass
```